File: validation-cpp-qt/src/data/repository/LocationRepository.cpp
/* ******************************************************************************
 * Copyright (c) 2025 Calypso Networks Association https://calypsonet.org/
 *
 * SPDX-License-Identifier: BSD-3-Clause
 ****************************************************************************** */

#include "LocationRepository.h"

namespace data::repository {

std::vector<domain::model::Location> LocationRepository::getLocations() const
{
    return initializeLocations();
}

std::vector<domain::model::Location> LocationRepository::initializeLocations() const
{
    // Same 12 European locations as Android app
    return {
        domain::model::Location(1, "Bruxelles"),
        domain::model::Location(2, "Konstanz"),
        domain::model::Location(3, "Lisboa"),
        domain::model::Location(4, "Milan"),
        domain::model::Location(5, "Munich"),
        domain::model::Location(6, "Paris"),
        domain::model::Location(7, "Riga"),
        domain::model::Location(8, "Roma"),
        domain::model::Location(9, "Strasbourg"),
        domain::model::Location(10, "Torino"),
        domain::model::Location(11, "Venice"),
        domain::model::Location(12, "Barcelona")
    };
}

} // namespace data::repository

File: src/validation/app/src/main/kotlin/org/calypsonet/keyple/demo/validation/data/LocationRepository.kt
/* ******************************************************************************
 * Copyright (c) 2021 Calypso Networks Association https://calypsonet.org/
 *
 * See the NOTICE file(s) distributed with this work for additional information
 * regarding copyright ownership.
 *
 * This program and the accompanying materials are made available under the
 * terms of the BSD 3-Clause License which is available at
 * https://opensource.org/licenses/BSD-3-Clause.
 *
 * SPDX-License-Identifier: BSD-3-Clause
 ****************************************************************************** */
package org.calypsonet.keyple.demo.validation.data

import javax.inject.Inject
import org.calypsonet.keyple.demo.validation.domain.model.Location

class LocationRepository @Inject constructor() {

  private val locationList: List<Location> =
      listOf(
          Location(0, "Bruxelles"),
          Location(1, "Konstanz"),
          Location(2, "Lisboa"),
          Location(3, "Milan"),
          Location(4, "Munich"),
          Location(5, "Paris"),
          Location(6, "Riga"),
          Location(7, "Roma"),
          Location(8, "Strasbourg"),
          Location(9, "Torino"),
          Location(10, "Venice"),
          Location(11, "Barcelona"))

  fun getLocations(): List<Location> {
    return locationList
  }
}
